module/data/activity/attack-data.mjs

```diff
@@ -60,6 +60,30 @@ export default class AttackActivityData extends BaseActivityData {
     return `${type.value === "ranged" ? "r" : "m"}${type.classification === "spell" ? "sak" : "wak"}`;
   }
 
+  /* -------------------------------------------- */
+
+  /**
+   * Abilities that could potentially be used with this attack. Unless a specific ability is specified then
+   * whichever ability has the highest modifier will be selected when making an attack.
+   * @type {Set<string>}
+   */
+  get availableAbilities() {
+    // Defer to item if available
+    if ( this.item.system.availableAbilities ) return this.item.system.availableAbilities;
+
+    // Spell attack not associated with a single class, use highest spellcasting ability on actor
+    if ( this.attack.type.classification === "spell" ) return new Set([
+      this.parent?.actor?.system.attributes?.spellcasting,
+      ...Object.values(this.parent?.actor?.spellcastingClasses ?? {}).map(c => c.spellcasting.ability)
```

Comment:
This implies that we would use the highest of the default spellcasting modifier or any spellcasting class' spellcasting modifier. There's no point in setting a default spellcasting modifier if it's just going to be ignored in that case.